File: plugins/codebase-review/references/ast-grep-starter-rules/test-fixtures/silent-failures/catch-only-logs-in-loop/negative.ts
// Fixture for catch-only-logs-in-loop — negative cases

declare const items: any[];
declare function processItem(x: any): Promise<any>;

// --- Case 1: Loop with failed[] array — correct pattern ---
async function correctWithFailed() {
  const results: any[] = [];
  const failed: any[] = [];
  for (const item of items) {
    try {
      results.push(await processItem(item));
    } catch (err) {
      failed.push({ id: item.id, error: (err as Error).message });
    }
  }
  return { results, failed };
}

// --- Case 2: Try/catch not inside a loop ---
async function singleTry(item: any) {
  try {
    return await processItem(item);
  } catch (err) {
    console.error(err);
    return null;
  }
}

// --- Case 3: Loop with re-throw in catch ---
async function loopRethrow() {
  const results: any[] = [];
  for (const item of items) {
    try {
      results.push(await processItem(item));
    } catch (err) {
      console.error(err);
      throw err;
    }
  }
  return results;
}

// --- Case 4: Loop with no try/catch ---
async function loopNoCatch() {
  const results: any[] = [];
  for (const item of items) {
    results.push(await processItem(item));
  }
  return results;
}

File: plugins/codebase-review/references/ast-grep-starter-rules/test-fixtures/silent-failures/python-except-log-only-in-loop/negative.py
# Fixture for python-except-log-only-in-loop — negative cases
import logging

logger = logging.getLogger(__name__)


def case_1_with_failed_list(items):
    """Correct pattern — parallel failed[] list."""
    results = []
    failed = []
    for item in items:
        try:
            results.append(process(item))
        except Exception as e:
            failed.append({"id": item.id, "error": str(e)})
    return {"results": results, "failed": failed}


def case_2_reraise(items):
    """Loop that logs and re-raises — not silent."""
    results = []
    for item in items:
        try:
            results.append(process(item))
        except Exception as e:
            logger.error("item %s failed: %s", item, e)
            raise
    return results


def case_3_try_not_in_loop(item):
    """Single try/except outside a loop — different pattern."""
    try:
        return process(item)
    except Exception as e:
        logger.error("failed: %s", e)
        return None


def case_4_loop_no_try(items):
    """Loop with no try/except — not the pattern."""
    return [process(item) for item in items]


def process(item):
    return item
